### DISCUSSION TITLE
Ability to have multiple workspaces on monitors

### DISCUSSION THREAD

**Discussion by rpensek** [nikitabobko/AeroSpace#663] [questions-and-answers]
Coming from Amethyst and switched to AeroSpace mainly for stability and performance improvements. Love the solution so far, the only thing that I'm missing here (or I'm unable to find it) is ability to have multiple workspaces for each monitor and toggle between them in a focused monitor using keyboard shortcuts. Is there a way to achieve this via settings, or is it by design to only support one workspace for each external monitor and support multiple workspaces only on main monitor?
I'd usually fallback to MacOS native multiple desktops (spaces) solution, but it doesn't appear to work fine with AeroSpace.

**Comment by alextremblay** [comment]
from my experience, aerospace actually makes this pretty easy with the `workspace-to-monitor-force-assignment` config option:

```
[workspace-to-monitor-force-assignment]
# You can assign workspaces to monitors by their names or by their patterns.
# Monitor pattern is a case-insensitive substring of the monitor name.
code = ['lg', 'secondary']
web = ['lg', 'secondary']
other = ['lg', 'secondary']
chat = 'dell'
reference = 'dell'
mail = 'built-in'
```
my 'built-in' monitor only has 1 workspace on it by default (mail), 
my 'lg' monitor has 3 workspaces on it: "code", "web", and "other",
and my 'dell' monitor has 2 workspaces on it: "chat" and "reference"

`workspace next/prev` and other workspace-related command function exactly how you would want them to, they operate on all the workspaces assigned to the currently-focused monitor
